src/logic/graphParsing.ts

```diff
@@ -30,10 +27,9 @@ export function isTriangle(graph: Graph): boolean {
 }
 
 export function isValid(text: string): boolean {
-  const graph = split(text);
-  if (graph.length === 0) return false;
-  if (!isSquare(graph) && !isTriangle(graph)) return false;
-  return true;
+  const graph = toMatrix(text);
+  if (!graph.length) return false;
```

Comment:
This is just a college project so allright. If it would've been commercial, I would not let it go. Why simply not use `else`? That way the code is more explicit, more extendable and more natural.

Here some source why to avoid this:
https://softwareengineering.stackexchange.com/questions/288398/when-to-use-single-line-if-statements
https://medium.com/@jonathanabrams/single-line-if-statements-2565c62ff492

... and more explanations here
https://helion.pl/ksiazki/czysty-kod-podrecznik-dobrego-programisty-robert-c-martin,czykov.htm#format/d

Reply:
In case of stack thread:
There is an answer that explains that one line ifs are not advised, but other answers suggest otherwise.

As for Your second link:
Case here is much more defined. The thing to notice is author's point. He doesn't state that one-liners are bad, but that using one-line-like syntax (without braces) in if with line break should be avoided.
In our case the problematic code (visible on second picture) won't compile. Of course, We shouldn't rely on our setup for that so I would still advise against it.

As for the book:
I am humble owner of physical copy of that book. I really like the author. It should be noted that this book doesn't really touch on code style in general, even despite it's name. Some time in the past I also watched some of his lectures and I haven't noticed him making a case against such practices.

The trick I used in line 31 is what's known as guard statement. Using `else` would defeat the purpose of it. Not only that, but would increase indentation and reduce code readability in rest of the function. Guard statements are encouraged to be used in the begging of functions, since reader after basic check can "dismiss" this part of code and get to the "meat" of function. More reading here:
[Medium - Python](https://medium.com/lemon-code/guard-clauses-3bc0cd96a2d3)
[Wikipedia](https://en.wikipedia.org/wiki/Guard_(computer_science))

And of course less lines != better code. At least not always. That's why I kept it on 2 lines instead of accepting Your  proposed 1 line solution.

Reply:
> In case of stack thread:
> There is an answer that explains that one line ifs are not advised, but other answers suggest otherwise.
Yes, but look which answers are the most upvoted ones.

> As for Your second link:
> Case here is much more defined. 
I just wanted to show you the situation from real life where this may cause trouble. The  second answer from stack gives more examples.

> The trick I used in line 31 is what's known as guard statement. Using else would defeat the purpose of it.
Medium link: those snippets in general are mess. One should strive to avoid putting multiple if in function body by splitting the logic into more levels of abstractions
Wikipedia: Even the example applies curly-braced approach. It's clean and explicit:
```
public string Foo(string username) {
    if (username == null) {
        throw new ArgumentNullException(nameof(username));
    }
    // Rest of the method code follows here...
}
```

Reply:
I don't have anything to write that I haven't wrote before, but I'll will leave it with a thought. There is a reason why this language feature didn't die out migrating from old C to Java, C#, TypeScript, Swift(with is relatively new), Rust (with is even newer), Python (although it is advised against here in most cases).
There is plenty of other examples on the web that don't use curly braces. The Wikipedia example is braced probably because it would look weird without rest of the function.
> One should strive to avoid putting multiple if in function body
It was used to demonstrate a point. Not to encourage writing that many ifs.

Reply:
> The Wikipedia example is braced probably because it would look weird without rest of the function.

Yes, bc this practice is just weird. Let's not lose time on that -  it's just a project.